app/src/main/java/com/davidtomas/taskyapp/MainActivityViewModel.kt

```diff
@@ -18,19 +18,24 @@ class MainActivityViewModel(
     val isAuthChecked = _isAuthChecked.asStateFlow()
 
     init {
-        authenticate()
+        viewModelScope.launch {
+            authenticate()
+        }
     }
 
     private fun authenticate() {
         viewModelScope.launch {
             authenticateUseCase().fold(
                 onError = {
-                    println(it)
+                    _isAuthenticated.value = false
+                    Any()
```

Comment:
Why this Any?

Reply:
Yes I think is a very dirty solution. The problem is that this two lambdas return an Any should I change them to Unit?. I dont understand what may be the advantages and disadvantages of using each of them
```
  inline fun fold(
        onError: (E) -> Unit or Any,
        onSuccess: (D) -> Unit or Any
    ): Any =
        when (this) {
            is Error -> onError(error)
            is Success -> onSuccess(data)
    }
```

Reply:
No point in returning Any. All functions without an explicit return type already return Unit, that's what Unit was made for